packages/react-server-dom-webpack/src/ReactFlightWebpackPlugin.js

```diff
@@ -94,6 +95,51 @@ export default class ReactFlightWebpackPlugin {
       options.manifestFilename || 'react-client-manifest.json';
   }
 
+  get webpackVersion(): 'wp5' | 'wp4' {
+    return require('webpack').version.match(/4\.[0-9]*\.[0-9]*/)
+      ? 'wp4'
+      : 'wp5';
+  }
+
+  pluginOutput = {};
+
+  recordModule = (
+    id: string,
+    {
+      getExports,
+      resource,
+      chunkIds,
+    }: {getExports: () => string[], resource: string, chunkIds: string[]},
+  ) => {
+    // TODO: Hook into deps instead of the target module.
+    // That way we know by the type of dep whether to include.
+    // It also resolves conflicts when the same module is in multiple chunks.
+    if (!/\.client\.(tsx|jsx|js|ts)$/.test(resource)) {
```

Comment:
From reading the RFC, I think the valid extensions should be any file that webpack accepts? EG: `resolve.extensions` 

In other words, from my undertanding of the RFCS: This code should be valid. 
```
// // Parent.server.js

import Component from "Component.client.js";
import implA from "implA.client.js";
import Picture from "MyCatPhoto.client.png";

export default function Parent() {
  return <Component avatar={Picture} impl={implA} />
}
```

(from: https://github.com/reactjs/rfcs/blob/235f27c12aa893efd2378ec3c4a9b0b221641861/text/0000-server-module-conventions.md#implicit-client-references